src/actions/security-chat.js

```diff
@@ -59,19 +103,50 @@ export function chatMessageIsTyping(boolean) {
     type: CHAT_MESSAGE_IS_TYPING,
   };
 }
-// export function chatRemoveMessage(messageId) {
-//   return {
-//     messageId: action.messageId,
-//     type: CHAT_REMOVE_MESSAGE,
-//   };
-// }
+export function chatRemoveMessage(messageId) {
+  removeMessage(messageId);
+  return {
+    messageId,
+    type: CHAT_REMOVE_MESSAGE,
+  };
+}
+export function updateMessage(messageId, updateMessageText) {
+  return {
+    messageId,
+    message: updateMessageText,
+    isTyping: false,
+    type: CHAT_UPDATE_MESSAGE_BY_ID,
+  };
+}
+export function fetchUpdateMessage(messageId, updateMessageText) {
+  return async dispatch => {
+    dispatch(fetchDataStart());
+    try {
+      const response = await fetch(
+        `http://localhost:8080/api/message/${messageId}`,
+        {
+          method: 'put',
+          body: JSON.stringify(updateMessageText),
```

Comment:
is `updateMessageText` a string or an object?

Reply:
object:
 const updateMessageText = {
        ...oldMessage,
        text: userInput,
      };

Reply:
Maybe I rename it to withUpdatedMessageText

Reply:
when it's called "text", we expect it to be string. rename it to be simply `messageUpdate`.